### ISSUE TITLE
missing TypeScript declaration file

### ISSUE THREAD

**Issue by Zamiell** [sindresorhus/eslint-plugin-unicorn#2324]
Now that ESLint 9 is released and everyone is migrating to flat config, importing the plugin into a flat config by following the official documentation actually results in a red squiggly line, because no declaration file exists (when using `checkJs` and/or `@ts-check`).

I suspect many/most users will be using TypeScript to validate that their new flat configs do not have typos in them. Thus, shipping a declaration file for this plugin would really be appreciated!

**Comment by TheElegantCoding** [comment]
same here 
![image](https://github.com/sindresorhus/eslint-plugin-unicorn/assets/70770072/a44525b9-4266-409f-b525-fc3a15423da0)

**Comment by fisker** [comment]
Types already available https://github.com/eslint-types/define-config-plugin-types/tree/main/types/unicorn

Maybe @Shinigami92 can help us move to this repo?

**Comment by silverwind** [comment]
Here's a workaround, but it's not much use because it does not validate `rules`. In fact I'm not sure how `rules` validation could even be made to work in typescript.

```ts
declare module "eslint-plugin-unicorn" {
  import type {Eslint} from "eslint";
  const plugin: Eslint.Plugin;
  export = plugin;
}
```
